src/main/java/teammates/ui/controller/Action.java

```diff
@@ -452,12 +452,38 @@ public String getRequestParamValue(String paramName) {
     }
     
     /**
+     * Retrieves the value for the specified parameter expected to be present in the http request.
+     *
+     * @param paramName  a constant from the {@link Const.ParamsNames} class.
+     * @return the value of the specified parameter.
+     * @throws NullPostParameterException if the parameter is not present in the http request.
+     */
+    public String getNonNullRequestParamValue(String paramName) {
+        String value = getRequestParamValue(paramName);
+        Assumption.assertPostParamNotNull(paramName, value);
+        return value;
+    }
+    
+    /**
      * @return null if the specified parameter was not found in the request.
      */
     public String[] getRequestParamValues(String paramName) {
         return HttpRequestHelper.getValuesFromParamMap(requestParameters, paramName);
     }
     
+    /**
+     * Retrieves the values for the specified parameter expected to be present in the http request.
+     *
+     * @param paramName  a constant from the {@link Const.ParamsNames} class.
+     * @return the values of the specified parameter.
+     * @throws NullPostParameterException if the parameter is not present in the http request.
+     */
+    public String[] getNonNullRequestParamValues(String paramName) {
+        String[] values = getRequestParamValues(paramName);
+        Assumption.assertPostParamNotNull(paramName, values);
+        return values;
+    }
+    
```

Comment:
> What about renaming Action to ControllerAction, have AutomatedAction and ControllerAction inherit from a new class Action? Besides these 2 methods, there are a few common methods and variables that can be placed in Action?

Better get @wkurniawan07 's input on this one. The de-duplication can be done as a separate PR too.